client/controller/index.web.js

```diff
@@ -24,25 +23,64 @@ export { setSection, setUpLocale } from './shared.js';
 
 const user = userFactory();
 
-export const ReduxWrappedLayout = ( { store, primary, secondary, redirectUri } ) => (
-	<ReduxProvider store={ store }>
-		{ getCurrentUser( store.getState() )
-			? <Layout primary={ primary }
-				secondary={ secondary }
-				user={ user }
-				nuxWelcome={ nuxWelcome }
-				translatorInvitation={ translatorInvitation }
-			/>
-			: <LayoutLoggedOut
-				primary={ primary }
-				secondary={ secondary }
-				redirectUri={ redirectUri }
-			/>
+/**
+ * Fetch the correct Layout based on whether or not the user is logged-in.
+ *
+ * @param {bool} loggedIn true if the user is logged-in, false otherwise
+ * @returns {Promise} a Promise that resolves to the Layout component
+ */
+function getLayout( loggedIn ) {
+	return new Promise( ( resolve ) => {
+		if ( loggedIn ) {
+			// If user is logged in, async load the main layout
+			require.ensure( [], () => {
```

Comment:
I tried using that first. The problem, as I understand it, is that many components expect to find the `#primary` or `#secondary` element on the page to render themselves into. So I needed to defer their rendering until those elements were available, and it's the `Layout` component that creates them.

I did this by defering the render through middleware. I don't think this would be possible with `AsyncLoad`. But I could be wrong! 😄